x/evm/keeper/abci.go

```diff
@@ -22,11 +22,12 @@ func (k *Keeper) BeginBlock(ctx sdk.Context, req abci.RequestBeginBlock) {
 	ctx = ctx.WithGasMeter(sdk.NewInfiniteGasMeter())
 
 	// Set the hash -> height and height -> hash mapping.
-	hash := req.Header.LastBlockId.GetHash()
-	height := req.Header.GetHeight() - 1
+	hash := req.Hash
+	height := req.Header.GetHeight()
```

Comment:
this should stay as it was before, the reason we are setting the hash for the previous block is because the block hash for the current height has not yet been determined (as we are just beginning the block).

Reply:
1. But we found that req.Hash is the block hash of the current height. Through this hash, you can go to the api "func (api *PublicEthereumAPI) GetBlockByHash" to successfully query the block information
2. Maybe the calculation of block hash only needs transaction and does not need the calculation result of transaction. Therefore, before beginBlock, the consensus of the block has ended, and the block hash is already determined